src/controlplane/controlplane_test.go

```diff
@@ -0,0 +1,310 @@
+package controlplane_test
+
+// This file holds the integration tests for the KSM package.
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/newrelic/nri-kubernetes/v2/internal/config"
+	"github.com/newrelic/nri-kubernetes/v2/internal/testutil"
+	"github.com/newrelic/nri-kubernetes/v2/src/controlplane"
+	"github.com/newrelic/nri-kubernetes/v2/src/definition"
+	"github.com/newrelic/nri-kubernetes/v2/src/metric"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+var (
+	excludeCM   = []string{"workqueueAddsDelta", "workqueueDepth", "workqueueDepth", "workqueueRetriesDelta"}
+	excludeETCD = []string{"processFdsUtilization"}
+	excludeS    = []string{"restClientRequestsDelta", "restClientRequestsRate", "schedulerScheduleAttemptsDelta", "schedulerScheduleAttemptsRate", "schedulerSchedulingDurationSeconds"}
+	excludeAS   = []string{"apiserverRequestsDelta", "apiserverRequestsRate", "restClientRequestsDelta", "restClientRequestsRate", "etcdObjectCounts"}
+)
+
+const masterNodeName = "masterNode"
+
+func Test_Scraper_Autodiscover_all_cp_components(t *testing.T) {
+	t.Parallel()
+
+	// Create an asserter with the settings that are shared for all test scenarios.
+	controlPlainSpecs := definition.SpecGroups{}
+	controlPlainSpecs["controller-manager"] = metric.ControllerManagerSpecs["controller-manager"]
+	controlPlainSpecs["etcd"] = metric.EtcdSpecs["etcd"]
+	controlPlainSpecs["scheduler"] = metric.SchedulerSpecs["scheduler"]
+	controlPlainSpecs["api-server"] = metric.APIServerSpecs["api-server"]
+
+	asserter := testutil.NewAsserter().
+		Using(controlPlainSpecs).
+		Excluding(
+			testutil.ExcludeMetrics("controller-manager", excludeCM...),
+			testutil.ExcludeMetrics("etcd", excludeETCD...),
+			testutil.ExcludeMetrics("scheduler", excludeS...),
+			testutil.ExcludeMetrics("api-server", excludeAS...),
```

Comment:
Why do we exclude this? I'm sure we have good reasons to, but cannot figure out which. Could we leave a comment explaining them?